bin/sozo/src/commands/build.rs

```diff
@@ -155,21 +155,44 @@ impl BuildArgs {
         // directly during the compilation to get the data we need from it.
         config.tokio_handle().block_on(bindgen.generate(None)).expect("Error generating bindings");
 
-        if self.stats != StatOptions::default() {
-            let world = WorldLocal::from_directory(
-                ws.target_dir_profile().to_string(),
-                ws.load_profile_config().unwrap(),
-            )?;
-
-            let world_stat = world.to_stat_item();
-            let mut stats = vec![world_stat];
-
-            for r in world.resources.values() {
-                if r.resource_type() != ResourceType::Namespace {
-                    stats.push(r.to_stat_item());
-                }
+        let world = WorldLocal::from_directory(
+            ws.target_dir_profile().to_string(),
+            ws.load_profile_config().unwrap(),
+        )?;
+        let world_stat = world.to_stat_item();
+        let mut stats = vec![world_stat];
+        for r in world.resources.values() {
+            if r.resource_type() != ResourceType::Namespace {
+                stats.push(r.to_stat_item());
+            }
+        }
+
+        // Check limits and emit warnings for EVERY compilation
+        for stat in &stats {
+            // Sierra file size check (4,089,446 bytes)
+            if stat.sierra_file_size > 4_089_446 {
```

Comment:
Could we move those magic values to constants? They are used below in the code, they may be extracted from the `from` implementation and put at the top of this module (with a link to the sn doc that mentioned those limits).